pkg/metrics/metrics.jsx

```diff
@@ -1518,18 +1516,20 @@ const PCPConfig = ({ buttonVariant, firewalldRequest, needsLogout, setNeedsLogou
         return null;
 
     function show_dialog() {
+        setDialogStatus(null);
         Dialogs.show(<PCPConfigDialog firewalldRequest={firewalldRequest}
-                                      needsLogout={needsLogout} setNeedsLogout={setNeedsLogout}
                                       s_pmlogger={s_pmlogger}
                                       s_pmproxy={s_pmproxy}
-                                      s_redis={s_redis} s_redis_server={s_redis_server} s_valkey={s_valkey} />);
+                                      s_redis={s_redis} s_redis_server={s_redis_server} s_valkey={s_valkey}
+                                      closeCallback={() => setDialogStatus("closed")} />);
     }
 
     return (
         <Button variant={buttonVariant} icon={<CogIcon />}
                 isDisabled={ invalidService(s_pmlogger) || invalidService(s_pmproxy) ||
                              invalidService(s_redis) || invalidService(s_redis_server) || invalidService(s_valkey) }
-                onClick={show_dialog}>
+                onClick={show_dialog}
+                data-test-dialog-closed={dialogStatus}>
```

Comment:
nit: `dialog-closed="closed"` feels a bit weird, especially if the value is something else. Maybe make it a boolean, or `dialog-status="closed"`?